app/org/sagebionetworks/bridge/dynamodb/DynamoExternalIdDao.java

```diff
@@ -249,9 +226,16 @@ public void deleteExternalIds(StudyIdentifier studyId, List<String> externalIds)
                     .withComparisonOperator(BEGINS_WITH));
         }
         if (assignmentFilter != null) {
-            addAssignmentFilter(query, assignmentFilter.booleanValue());
+            query.withQueryFilterEntry(HEALTH_CODE, new Condition()
+                .withComparisonOperator(assignmentFilter.booleanValue() ? NOT_NULL : NULL));
+        }
+        if (callerSubstudyIds != null && !callerSubstudyIds.isEmpty()) {
+            query.withQueryFilterEntry(SUBSTUDY_ID, new Condition()
+                .withAttributeValueList(callerSubstudyIds.stream()
+                        .map(id -> new AttributeValue(id)).collect(Collectors.toList()))
+                .withComparisonOperator(ComparisonOperator.IN));
```

Comment:
TODO: I'm not sure how to get the benefits of a secondary index without an entirely different query on the index... the same could be said of the health code assignment query filter. I'd have to think about how this would work. Note that this query is only for administrators who are looking at lists of external IDs... it's not used during authentication, etc.